File: backend/app/api/authentication/schemas/employee.py
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, EmailStr, ConfigDict
from enum import Enum


class EmployeeRoleEnum(str, Enum):
    OWNER = "owner"
    ADMIN = "admin"
    USER = "user"


class EmployeeStatusEnum(str, Enum):
    PENDING = "pending"
    ACTIVE = "active"
    INACTIVE = "inactive"
    DELETED = "deleted"


class EmployeeBase(BaseModel):
    email: EmailStr
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    patronymic: Optional[str] = None
    phone: Optional[str] = None
    position: Optional[str] = None
    role: EmployeeRoleEnum = EmployeeRoleEnum.USER
    permissions: Optional[Dict[str, bool]] = None


class EmployeeCreate(BaseModel):
    email: EmailStr
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    patronymic: Optional[str] = None
    phone: Optional[str] = None
    position: Optional[str] = None
    role: EmployeeRoleEnum = EmployeeRoleEnum.USER
    permissions: Optional[Dict[str, bool]] = None


class EmployeeUpdate(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    patronymic: Optional[str] = None
    phone: Optional[str] = None
    position: Optional[str] = None
    role: Optional[EmployeeRoleEnum] = None
    permissions: Optional[Dict[str, bool]] = None


class EmployeeResponse(EmployeeBase):
    id: int
    user_id: Optional[int] = None
    company_id: int
    status: EmployeeStatusEnum
    deletion_requested_at: Optional[datetime] = None
    deletion_requested_by: Optional[int] = None
    deletion_rejected_at: Optional[datetime] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    created_by: Optional[int] = None
    
    model_config = ConfigDict(from_attributes=True)


class EmployeeListResponse(BaseModel):
    employees: List[EmployeeResponse]
    total: int
    page: int
    per_page: int


class PermissionUpdateRequest(BaseModel):
    permissions: Dict[str, bool]


class AdminDeletionRequest(BaseModel):
    reason: Optional[str] = None


class AdminDeletionRejectRequest(BaseModel):
    reason: Optional[str] = None


# Список всех доступных прав в системе
AVAILABLE_PERMISSIONS = {
    "company_management": "Управление компанией",
    "company_data": "Данные компании", 
    "products": "Продукция",
    "announcements": "Объявления",
    "business_connections": "Бизнес-связи",
    "partners": "Партнеры",
    "suppliers": "Поставщики", 
    "buyers": "Покупатели",
    "documents": "Документы",
    "contracts": "Договоры",
    "sales": "Продажи",
    "purchases": "Закупки",
    "communications": "Коммуникации",
    "messages": "Сообщения",
    "administration": "Администрирование"
}

File: frontend/types/employee.ts
export interface EmployeeResponse {
  id: number
  user_id?: number
  company_id: number
  email: string
  first_name?: string
  last_name?: string
  patronymic?: string
  phone?: string
  position?: string
  role: 'owner' | 'admin' | 'user'
  status: 'pending' | 'active' | 'inactive' | 'deleted'
  permissions: Record<string, boolean>
  deletion_requested_at?: string
  deletion_requested_by?: number
  deletion_rejected_at?: string
  created_at: string
  updated_at?: string
  created_by?: number
}

export interface EmployeeCreate {
  email: string
  first_name?: string
  last_name?: string
  patronymic?: string
  phone?: string
  position?: string
  role: 'owner' | 'admin' | 'user'
  permissions?: Record<string, boolean>
}

export interface EmployeeUpdate {
  first_name?: string
  last_name?: string
  patronymic?: string
  phone?: string
  position?: string
  role?: 'owner' | 'admin' | 'user'
  permissions?: Record<string, boolean>
}

export interface EmployeeListResponse {
  employees: EmployeeResponse[]
  total: number
  page: number
  per_page: number
}

export interface PermissionUpdateRequest {
  permissions: Record<string, boolean>
}

export interface AdminDeletionRequest {
  reason?: string
}

export interface AdminDeletionRejectRequest {
  reason?: string
}

// Список всех доступных прав в системе
export const AVAILABLE_PERMISSIONS = {
  company_management: "Управление компанией",
  company_data: "Данные компании", 
  products: "Продукция",
  announcements: "Объявления",
  business_connections: "Бизнес-связи",
  partners: "Партнеры",
  suppliers: "Поставщики", 
  buyers: "Покупатели",
  documents: "Документы",
  contracts: "Договоры",
  sales: "Продажи",
  purchases: "Закупки",
  communications: "Коммуникации",
  messages: "Сообщения",
  administration: "Администрирование"
} as const
